src/garage/experiment/local_tf_runner.py

```diff
@@ -44,26 +64,17 @@ class LocalRunner:
 
     """
 
-    def __init__(self, sess=None, max_cpus=1):
-        """Create a new local runner.
-
-        Args:
-            max_cpus(int): The maximum number of parallel sampler workers.
-            sess(tf.Session): An optional tensorflow session.
-                  A new session will be created immediately if not provided.
-
-        Note:
-            The local runner will set up a joblib task pool of size max_cpus
-            possibly later used by BatchSampler. If BatchSampler is not used,
-            the processes in the pool will remain dormant.
-
-            This setup is required to use tensorflow in a multiprocess
-            environment before a tensorflow session is created
-            because tensorflow is not fork-safe.
+    def __init__(self, snapshot_config=None, sess=None, max_cpus=1):
+        if isinstance(snapshot_config, SnapshotConfig):
```

Comment:
i think you should restructure this using a guard clause:

https://blog.mafr.de/2009/06/12/a-case-for-guard-clauses/

```python
self.snapshotter = None
if snapshot_config:
    self.snapshotter = Snapshotter(...)
else:
    warnings.warn('No snapshotter created...etc.')
```